services/brig/src/Brig/App.hs

```diff
@@ -493,6 +494,21 @@ instance MonadReader Env (AppT r) where
   ask = AppT ask
   local f (AppT m) = AppT $ local f m
 
+instance Member (Input Env) r => MonadReader Env (Sem r) where
+  ask = input
+  local f sem = do
+    env <- input
+    runInputConst (f env) . raise @(Input Env) $ sem
+
+instance
+  ( Members '[Final IO, Embed IO, Embed Cas.Client] r,
```

Comment:
Why do you need IO here?

Reply:
I'm deep into compilation issues so I can't answer this off the bat. I think something needed it.

I'm thinking of closing this PR. First I've started cleaning up per your suggestion, but then I got into a rabbit hole that looked a lot like what I was already doing in a PR that builds on top of this, but that has more things polysemized.